File: src/org/kopi/vkopi/lib/ui/vaadin/addons/GridEditorEnumField.java
/*
 * Copyright (c) 1990-2017 kopiRight Managed Solutions GmbH
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 * $Id$
 */

package org.kopi.vkopi.lib.ui.vaadin.addons;

import org.kopi.vkopi.lib.ui.vaadin.addons.client.grid.EditorEnumFieldState;

/**
 * Server side implementation of an enumeration editor
 */
@SuppressWarnings("serial")
public class GridEditorEnumField extends GridEditorTextField {

  //---------------------------------------------------
  // CONSTRUCTOR
  //---------------------------------------------------
  
  public GridEditorEnumField(int width, String[] enumerations) {
    super(width);
    getState().enumerations = enumerations;
  }

  //---------------------------------------------------
  // IMPLEMENTATION
  //---------------------------------------------------
  
  @Override
  protected EditorEnumFieldState getState() {
    return (EditorEnumFieldState) super.getState();
  }
}

File: src/org/kopi/vkopi/lib/ui/vaadinflow/grid/GridEditorEnumField.kt
/*
 * Copyright (c) 2013-2022 kopiLeft Services SARL, Tunis TN
 * Copyright (c) 1990-2022 kopiRight Managed Solutions GmbH, Wien AT
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License version 2.1 as published by the Free Software Foundation.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
package org.kopi.vkopi.lib.ui.vaadinflow.grid

/**
 * Server side implementation of an enumeration editor
 */
class GridEditorEnumField(width: Int, private val enumerations: Array<String>?) : GridEditorTextField(width) {

  override fun check(text: String): Boolean {
    if (enumerations != null) {
      val s = text.lowercase()
      for (i in enumerations.indices) {
        if (enumerations[i].lowercase().startsWith(s)) {
          return true
        }
      }
      return false
    }
    return true
  }
}
